src/Math/src/AlgebraicRiccatiEquation.cpp

```diff
@@ -0,0 +1,129 @@
+                                                            Eigen::Ref<const Eigen::MatrixXd> R,
+                                                            double tolerance,
+                                                            std::size_t maxIterations,
+                                                            bool verbose)
+{
+    constexpr std::string_view errorPrefix = "[BipedalLocomotion::Math::"
+                                             "ContinuousAlgebraicRiccatiEquation] ";
+
+    std::pair<bool, Eigen::MatrixXd> result;
+
+    const std::size_t n = B.rows();
+    const std::size_t m = B.cols();
+
+    // check the size of the matrices
+    if (A.rows() != n || A.cols() != n)
+    {
+        std::cerr << errorPrefix << "The matrix A must be a square matrix. Expected size: " << n
+                  << " x " << n << ". Passed size: " << A.rows() << " x " << A.cols() << "."
+                  << std::endl;
+        result.first = false;
+        return result;
+    }
+
+    if (Q.rows() != n || Q.cols() != n)
+    {
+        std::cerr << errorPrefix << "The matrix Q must be a square matrix. Expected size: " << n
+                  << " x " << n << ". Passed size: " << Q.rows() << " x " << Q.cols() << "."
+                  << std::endl;
+        result.first = false;
+        return result;
+    }
+
+    if (R.rows() != m || R.cols() != m)
+    {
+        std::cerr << errorPrefix << "The matrix Q must be a square matrix. Expected size: " << n
+                  << " x " << n << ". Passed size: " << Q.rows() << " x " << Q.cols() << "."
+                  << std::endl;
+        result.first = false;
+        return result;
+    }
+
+    Eigen::LLT<Eigen::MatrixXd> R_cholesky(R);
+    if (R_cholesky.info() != Eigen::Success)
+    {
+        std::cerr << errorPrefix << "The matrix R must be positive definite." << std::endl;
+        result.first = false;
+        return result;
+    }
+
+    constexpr double symmetricTolerance = 1e-8;
+    if (!Q.isApprox(Q.transpose(), symmetricTolerance))
+    {
+        std::cerr << errorPrefix << "The matrix Q must be symmetric." << std::endl;
+        result.first = false;
+        return result;
+    }
+
+    Eigen::MatrixXd H(2 * n, 2 * n);
+    H << A, B * R_cholesky.solve(B.transpose()),
+         Q, -A.transpose();
+
+    Eigen::MatrixXd Z = H;
+    Eigen::MatrixXd Z_old;
+
+    double relativeNorm = tolerance;
+    std::size_t iteration = 0;
+    const double p = static_cast<double>(Z.rows());
+
+    for (; iteration < maxIterations && relativeNorm >= tolerance;
+         iteration++)
+    {
+        Z_old = Z;
+        // R. Byers. Solving the algebraic Riccati equation with the matrix sign
+        // function. Linear Algebra Appl., 85:267–279, 1987
+        // Added determinant scaling to improve convergence (converges in rough half
+        // the iterations with this)
+        const double ck = std::pow(std::abs(Z.determinant()), -1.0 / p);
```

Comment:
```suggestion
        const double ck = std::pow(std::abs(Z.determinant()), -1.0 / (2*p));
```

Looks like that in the original paper?

Reply:
I think the above comment can be safely ignored. For the implemented one is the good, see Eqn 2. of the paper.